lib/mongoid/traversable.rb

```diff
@@ -17,6 +17,25 @@ def _parent=(p)
       @__parent = p
     end
 
+    included do 
+      module DiscriminatorKeyAssignment
+        def discriminator_key=(value)
+          if hereditary?
+            raise Errors::SettingDiscriminatorKeyOnChild.new(self.to_s, self.superclass.to_s)
+          end
+
+          value = Mongoid.discriminator_key unless value
```

Comment:
This should remove the override instead of writing the global value in.

Reply:
Would you know how to achieve this? I've been googling but I can't figure out how to remove the override (or redelegate to Mongoid)...

Things I've tried: 
`delegate :discriminator_key, to: ::Mongoid unless value`

Reply:
I think my implementation, while isn't exactly what you would expect, is a better implementation... Here's the use case I'm thinking about. Imagine the following code was executed:
```
Instrument.discriminator_key = nil
Instrument.discriminator_key = nil
```
If we remove the override of the discriminator_key on the first line and redelegate `discriminator_key` to `Mongoid`, then the second line will set the global discriminator_key to nil, which is presumably not what we want. In my implementation, `Instrument.discriminator_key` will return the global discriminator key after each line.

Reply:
Resolved!